### ISSUE TITLE
Unable to size OTSubscriber without an explicit width / height and broken expectations

### ISSUE THREAD

**Issue by jeanregisser** [opentok/opentok-react-native#121]
It's currently impossible to size `<OTSubscriber />` without an explicit width / height.

The following example will not display anything:
```jsx
<View style={{ flex: 1 }}>
  <OTSubscriber
    sessionId={opentokSession.sessionId}
    style={{ flex: 1 }}
  />
</View>
```

The main reason is because the passed style is applied to the children instead of the container view:
```jsx
// OTSubscriber.js
render() {
  const childrenWithStreams = this.state.streams.map((streamId) => {
    const streamProperties = this.props.streamProperties[streamId];
    const style = isEmpty(streamProperties) ? this.props.style : isNull(streamProperties.style) ? this.props.style : streamProperties.style;
    return <OTSubscriberView key={streamId} streamId={streamId} style={style} />
  });
  return <View>{ childrenWithStreams }</View>;
}
```

Also note that it's impossible to apply any additional `ViewProps` to the rendered `<View>`.

One way to fix this is to apply the style to the container.
And have a default style (`flex: 1`) for the `<OTSubscriberView>` which could be overridden with a passed prop (maybe named streamViewStyle?).

For maximum compatibility, other props passed to the component should be passed to the container view as well.

Here's a full example of what I'd expect:

```jsx
render() {
  const {
    style,
    streamStyle,
    properties,
    eventHandlers,
    streamProperties,
    ...other
  } = this.props;
  const childrenWithStreams = this.state.streams.map(streamId => {
    const streamProps = streamProperties[streamId];
    return (
      <OTSubscriberView
        key={streamId}
        streamId={streamId}
        style={[streamStyle, streamProps && streamProps.style]}
      />
    );
  });
  return (
    <View style={style} {...other}>
      {childrenWithStreams}
    </View>
  );
}
```

The same ideas should be applied to the other components in the lib.

Also this is an interesting read: https://medium.freecodecamp.org/introducing-the-single-element-pattern-dfbd2c295c5d

**Comment by msach22** [comment]
@jeanregisser Thanks for reporting this, I'll investigate this further and update it as soon as possible.

Also, please keep in mind that you need to specify a width and height because the native view requires a width and height so the video can be rendered on the screen.

**Comment by msach22** [comment]
@jeanregisser I'm going to close this issue because the fix has been released with `v0.8.4`. Please let me know if you have any questions!
